### PR TITLE
Register task with Api integration. 

### LINKED ISSUE #5: User Registration
- Create a registration form with the listed fields.
- Add front end validation on the required fields and validate password and email format.
- Submit the registration form to the backend.
- Handle displaying any errors form the backend response.
- Handle redirecting to verify email message after registration.

### PR DESCRIPTION
**Pull request by AbdullahMuhammed5**
Acceptance criteria: 

- Create registration form with the listed fields.
- Add front end validation on the required fields and validate password and email format.
- Submit the registration form to the back-end.
- Handle displaying any errors form the back-end response
- Handle redirecting to verify email message after registration.

Here is a screen for the design:
screencapture-localhost-3000-Register-2020-09-05-16_17_51

and here is a screen for a passed tests: 
Screenshot from 2020-09-05 15-12-16

### CODE REVIEW

**Review by aashrafh — DISMISSED** [review]

**Review by mohamedsgap — CHANGES_REQUESTED** [review]
you are really did a great work, mate :+1:  just to mention here you need to pull the latest update from the `develop` branch especially after the `Create Offer` has merged and compare the changes to just fix the current conflict of the files. Also once you integrated with an API, you need to make sure that the form has been submitted  in the unit testing by simulating its behavior

**Comment by AbdullahMuhammed5** [comment]
> you are really did a great work, mate  just to mention here you need to pull the latest update from the `develop` branch especially after the `Create Offer` has merged and compare the changes to just fix the current conflict of the files. Also once you integrated with an API, you need to make sure that the form has been submitted in the unit testing by simulating its behavior

Thank you @mohamedsgap I did merge and resolve the conflicts.
Regarding unit tests I was stack in it so I decided to merge the task and work on unit test for Api later.

**Review by mohamedsgap — DISMISSED** [review]
So great bro :100: @AbdullahMuhammed5

**Review by aashrafh — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -5,6 +5,8 @@
   "dependencies": {
     "@date-io/date-fns": "^1.3.9",
     "@material-ui/core": "^4.11.0",
+    "axios": "^0.20.0",
+    "bootstrap": "^4.3.1",
```
Comment (ghost) [inline package.json:9]: bootstrap is not needed, it should be removed.

**Review by mohamedsgap — APPROVED** [review]

**Review by mohamedsgap — APPROVED** [review]

**Review by aashrafh — APPROVED** [review]
Good work

**Review by ghost — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (mohamedsgap) [reply package.json:9]: I think some of MUI library is using it, @AbdullahMuhammed5 isn't it ?

**Pull request merged**